src/features/dialog-manager/reducer.js

```diff
@@ -165,10 +165,9 @@ const dialogManagerReducer = (state = initialState, { type, payload }) => {
             return {
                 ...state,
                 character: {
-                    ...character,
-                    characterName: payload.characterName,
-                    characterRace: payload.characterRace,
-                    characterClass: payload.characterClass,
+                    characterName: payload.characterName || characterName,
+                    characterRace: payload.characterRace || characterRace,
+                    characterClass: payload.characterClass || characterClass,
```

Comment:
What's the reasoning behind putting the '|| <whatever>' here? As far as I'm concerned, if the <whatever> is undefined, null, or something else 'falsey' then that should be a bug in our logic. I'm happy to hear your thoughts though, because if there's a good reason we should consider adding this to the majority of the reducers.

Good catch on the characterName though.